### PR TITLE
Added exception causes

### LINKED ISSUE #111: Report exception causes
Python supports chaining exceptions, when an exception occurs while another exception is already being handled:

https://docs.python.org/3/library/exceptions.html#exception-context

Our /v1/notices spec defines the exception cause [here](https://github.com/honeybadger-io/honeybadger-schemas/blob/77bcff997132d5393b51db26d1cc77efd6f0dc88/exceptions/v1/notices.json#L184).

Here's how our ruby gem collects and reports a chain of exception causes:

https://github.com/honeybadger-io/honeybadger-ruby/blob/bb19ded6e557e6e9713229d46ea5b891d2b6aeb3/lib/honeybadger/notice.rb#L198

https://github.com/honeybadger-io/honeybadger-ruby/blob/bb19ded6e557e6e9713229d46ea5b891d2b6aeb3/lib/honeybadger/notice.rb#L241

### PR DESCRIPTION
**Pull request by Kelvin4664**
image
Resolves #111

### CODE REVIEW

File: honeybadger/payload.py [hunk 1]
```diff
@@ -33,11 +39,8 @@ def is_not_honeybadger_frame(frame):
 
     logger.debug(tb)
 
-    payload = {
-        'class': type(exception) is dict and exception['error_class'] or exception.__class__.__name__,
-        'message': type(exception) is dict and exception['error_message'] or str(exception),
-        'backtrace': [dict(number=f[1], file=_filename(f[0]), method=f[2], source=read_source(f)) for f in reversed(tb)]
-    }
+    payload = prepare_exception_payload(exception)
+    payload['causes'] = [prepare_exception_payload(exception.__cause__) ] if hasattr(exception, '__cause__') else []
```
Comment (joshuap) [inline honeybadger/payload.py:43]: Could we also add recursive causes to this list? For example, check to see if `exception.__cause__` also has a `__cause__` attribute, and add it to the list if so (and do that for each cause, until there is no additional `__cause__`?

**Review by joshuap — CHANGES_REQUESTED** [review]
@Kelvin4664 good start, thanks! I added one comment. Also, could you add some tests and a [changelog entry](https://github.com/honeybadger-io/honeybadger-python/blob/master/CHANGELOG.md)?

File: honeybadger/payload.py (hunk 1 above)
Comment (Kelvin4664) [reply honeybadger/payload.py:43]: I thought of handling it recursively but wasn't sure if nested `causes` is supported by the notices spec.
I'll update.

Comment (joshuap) [reply honeybadger/payload.py:43]: Yep, the array can have as many causes as you want. They are displayed in order, so the most recent cause should be displayed first, and the oldest cause should be displayed last.

Comment (Kelvin4664) [reply honeybadger/payload.py]: All done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ CHANGELOG](http://keepachangelog.com/) for how to update this file. This project
 adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
-
+- Recursively add nested exceptions to exception 'causes'
```
Comment (joshuap) [inline CHANGELOG.md:7]:
```suggestion
### Added
- Recursively add nested exceptions to exception 'causes'
```

**Review by joshuap — APPROVED** [review]
Thanks!

**Comment by joshuap** [comment]
@subzero10 do you want to take a shot at merging/releasing this? I can get you set up to release on pypi.org if you create an account and Slack it to me.

**Comment by subzero10** [comment]
Done! https://pypi.org/project/honeybadger/0.9.0/

**Pull request merged**